lib/errors.js

```diff
@@ -1,17 +1,56 @@
-'use strict';
+const statuses = require('statuses');
 
-const util = require('util');
-
-function CustomError (message) {
-  Error.captureStackTrace(this, this.constructor);
-  this.name = this.constructor.name;
-  this.message = message;
+class CustomError extends Error {
+  constructor (message) {
+    super();
+    this.message = message;
+    this.name = this.constructor.name;
+    Error.captureStackTrace(this, this.constructor);
+  }
 }
-util.inherits(CustomError, Error);
 
 class ConfigurationError extends CustomError {}
 
+class GatewayError extends CustomError {
+  constructor (code = 500, errors = null) {
+    super();
+    this.code = 500;
+    this.error = null;
+    this.errors = [];
+    switch (typeof code) {
+      case 'number':
+        if (code >= 400 && code < 600) {
+          this.code = code;
+        }
+        break;
+      default:
+        errors = code;
+    }
+    if (Array.isArray(errors)) {
+      this.errors = errors.filter((error) => error && typeof error === 'string');
+    } else if (typeof errors === 'string') {
+      this.error = errors;
```

Comment:
can it be always `errors` as array. otherwise consumer will have to check 2 fields

Reply:
I'm thinking about changing it to:
```
{
  "errno": number,
  "error": string, 
  "data": any // as an option for eg. validation 
}
```